### PR TITLE
Support running multiple guardians

### PR DESCRIPTION
**Pull request by Lbqds**
1. Support running multiple guardians: when the contract init fields change, contract ids on Alephium will also change. In order to avoid modifying the config file, the `UpdateGuardianSet` governance action will be executed after the contract is deployed. You can start multiple guardians by running this command: `tilt up -- --num=GUARDIAN_NUM --webHost=0.0.0.0`
2. Sequence: VAA is uniquely identified by `emitter chain id + emitter address + target chain id + sequence`, `TokenBridge` and `Governance` contracts share the same `GovernanceEmitterChainId` and `GovernanceEmitterAddress`, so in order to ensure the uniqueness of the VAA key, I changed the governance sequence validation rules to non-continuous increments.
3. MessageFee: Now if you want to transfer `N` WALPH from Alephium to ETH, then you will receive `N - messageFee` WALPH on ETH, this is for consistency with ETH
4. Change the ETH chain id from 1337 to 1338 on devnet, because the latest metamask cannot import the custom tokens from chain id 1337, more info at [here](https://community.metamask.io/t/not-able-to-import-custom-tokens-on-mm-from-ganache/22939)

### CODE REVIEW

**Review by h0ngcha0 — COMMENTED** [review]
Finished first round of review, great work :+1:

File: alephium/contracts/governance.ral [hunk 1]
```diff
@@ -92,22 +92,22 @@ Contract Governance(
     }
 
     @using(readonly = true, externalCallCheck = false)
-    pub fn parseAndVerifyGovernanceVAAGeneric(vaa: ByteVec, targetSequence: U256, coreModule: U256, action: ByteVec) -> (U256, ByteVec) {
+    pub fn parseAndVerifyGovernanceVAAGeneric(vaa: ByteVec, targetSequence: U256, coreModule: U256, action: ByteVec) -> (U256, U256, ByteVec) {
         let (emitterChainId, targetChainId, emitterAddress, msgSequence, payload) = parseAndVerifyVAA(vaa, true)
         assert!(emitterChainId == governanceChainId, ErrorCodes.InvalidEmitChainId)
         assert!(emitterAddress == governanceEmitterAddress, ErrorCodes.InvalidEmitAddress)
-        assert!(msgSequence == targetSequence, ErrorCodes.InvalidSequence)
+        assert!(msgSequence >= targetSequence, ErrorCodes.InvalidSequence)
```
[the remark this answers is not in the document]
Comment (Lbqds) [reply alephium/contracts/governance.ral:99]: This is because the `Governance` contract and the `TokenBridge` contract use the same `GovernanceEmitterChainId` and `GovernanceEmitterAddress`, it is possible that the two vaas acting on the `Governance` contract and the `TokenBridge` contract respectively have the same vaa id, which will cause storage problems.

All governance messages are generated by off-chain guardians, so only +2/3 guardians signing a vaa with a large msgSequence would cause an attack.

File: alephium/test/governance-test.ts [hunk 2]
```diff
@@ -83,17 +84,24 @@ describe('test governance', () => {
...
+      )
+      const vaa = initGuardianSet.sign(initGuardianSet.quorumSize(), body)
+      return testCase(vaa, 'submitNewGuardianSet', undefined, undefined, 3)
+    }
+    for (let seq = 0; seq < 3; seq += 1) {
+      await expectAssertionFailed(async () => await test(seq))
+    }
+    for (let seq = 3; seq < 5; seq += 1) {
+      // we have checked the results in previous tests
+      await test(seq)
```
[the remark this answers is not in the document]
Comment (Lbqds) [reply alephium/test/governance-test.ts:103]: Sounds great, thanks!

File: clients/js/main.ts [hunk 3]
```diff
@@ -179,6 +179,58 @@ yargs(hideBin(process.argv))
...
+                  return key.slice(2)
+                }
+                return key
+              })
+              if (keys.length === 0) {
+                throw new Error('new guardian set cannot be empty')
+              }
+              const payload: vaa.GuardianSetUpgrade = {
+                module: 'Core',
+                type: 'GuardianSetUpgrade',
```
[the remark this answers is not in the document]
Comment (Lbqds) [reply clients/js/main.ts:218]: Yes, I will fix it to keep the name consistent, thanks!

File: node/pkg/db/db.go [hunk 4]
```diff
@@ -121,6 +125,41 @@ func (d *Database) StoreSignedVAA(v *vaa.VAA) error {
 	return nil
 }
 
+func (d *Database) MaxGovernanceVAASequence() (*uint64, error) {
```
[the remark this answers is not in the document]
Comment (Lbqds) [reply node/pkg/db/db.go:128]: Yes, I think it's fine because we won't have too many governance vaas.

File: node/cmd/guardiand/node.go [hunk 5]
```diff
@@ -640,10 +641,25 @@ func runNode(cmd *cobra.Command, args []string) {
...
+		logger.Info("wait for the bootstrap node to establish networking", zap.Int("index", *devnetGuardianIndex))
+		time.Sleep(15 * time.Second)
+	}
+
 	// In devnet mode, we generate a deterministic guardian key and write it to disk.
 	if *unsafeDevMode {
 		gk := devnet.InsecureDeterministicEcdsaKeyByIndex(ethcrypto.S256(), uint64(*devnetGuardianIndex))
-		err := writeGuardianKey(gk, "auto-generated deterministic devnet key", *guardianKeyPath, true)
+		keyPath := fmt.Sprintf("%s-%d", *guardianKeyPath, *devnetGuardianIndex)
+		guardianKeyPath = &keyPath
```
[the remark this answers is not in the document]
Comment (Lbqds) [reply node/cmd/guardiand/node.go:654]: Sounds great, thanks!

File: alephium/test/token-bridge-test.ts [hunk 6]
```diff
@@ -351,34 +359,35 @@ describe('test token bridge', () => {
     const tokenPoolState = testResult.contracts.filter(
       (c) => c.contractId === fixture.wrappedAlphPoolInfo.contractId
     )[0]
-    expect(tokenPoolState.fields['totalBridged']).toEqual(fixture.totalBridged + transferAmount)
+    const realTransferAmount = transferAmount - messageFee
```
Comment (polarker) [inline alephium/test/token-bridge-test.ts:362]: Do we test both zero fee and non-zero fee?

File: alephium/scripts/09_update_guardian_set.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { Project } from '@alephium/web3'
+import { Configuration, Deployer, DeployFunction, Network } from '@alephium/cli/types'
+import { Settings } from '../alephium.config'
+
+const updateGuardianSet: DeployFunction<Settings> = async (
```
[the remark this answers is not in the document]
Comment (Lbqds) [reply alephium/scripts/09_update_guardian_set.ts:5]: It is only for devnet. Sometimes we have to run multiple guardians, which means that the initial fields of the `Governance` contract are different, and the contract address will also be different. To avoid modifying configuration files, I added this script to upgrade the guardian set, then the contract address will not change.

File: alephium/test/token-bridge-test.ts (hunk 6 above)
Comment (Lbqds) [reply alephium/test/token-bridge-test.ts:362]: Now only non-zero fee, will add tests for zero fee, thanks!

File: alephium/scripts/09_update_guardian_set.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (polarker) [reply alephium/scripts/09_update_guardian_set.ts:5]: I see. We could also deploy the bridge multiple times instead of modifying the configuration files, but I am not sure it's convenient to do so.
